src/TLuaInterpreterMapper.cpp

```diff
@@ -577,18 +577,30 @@ int TLuaInterpreter::addRoom(lua_State* L)
     const int id = getVerifiedInt(L, __func__, 1, "roomID");
     const Host& host = getHostFromLua(L);
     const bool added = host.mpMap->addRoom(id);
-    lua_pushboolean(L, added);
+    bool issueBadAreaWarning = false;
     if (added) {
         int areaID = -1;
-        if (lua_gettop(L) >= 2) {
+        if (lua_gettop(L) > 1) {
             areaID = getVerifiedInt(L, __func__, 2, "areaID");
         }
         // defer area calculations as all new rooms are initialised at 0,0,0 anyway
-        host.mpMap->setRoomArea(id, areaID, true);
-        host.mpMap->setUnsaved(__func__);
```

Comment:
This and the `TMap::mMapGraphNeedsUpdate` are not required here as they have been carried out by the methods already called.